fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -173,21 +173,25 @@ def app_name
     end
 
     def mac?
-      # Some projects have different values... we have to look for all of them
-      return true if build_settings(key: "PLATFORM_NAME") == "macosx"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "macOS"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "OS X"
-      false
+      supported_platforms.include?(:macOS)
     end
 
     def tvos?
-      return true if build_settings(key: "PLATFORM_NAME").to_s.include? "appletv"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME").to_s.include? "tvOS"
-      false
+      supported_platforms.include?(:tvOS)
     end
 
     def ios?
-      !mac? && !tvos?
+      supported_platforms.include?(:iOS)
+    end
+
+    def supported_platforms
+      supported_platforms = build_settings(key: "SUPPORTED_PLATFORMS").split
+      platforms = []
+      platforms << :macOS   if supported_platforms.include?("macosx")
```

Comment:
What about the others that were previously in the `mac?` method? Or does the `SUPPORTED_PLATFORMS` key always just contain `macosx`? 

```
return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "macOS"
return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "OS X"
```